tests/test_tools_pr_comments.py

```diff
@@ -0,0 +1,86 @@
+# Tests for functions defined in 'tools/pr_comments.py' of the EESSI
+# build-and-deploy bot, see https://github.com/EESSI/eessi-bot-software-layer
+#
+# The bot helps with requests to add software installations to the
+# EESSI software layer, see https://github.com/EESSI/software-layer
+#
+# author: Thomas Roeblitz (@trz42)
+#
+# license: GPLv2
+#
+
+# Standard library imports
+
+# Third party imports (anything installed into the local Python environment)
+import pytest
+from unittest.mock import patch
+
+# Local application imports (anything from EESSI/eessi-bot-software-layer)
+from tools.pr_comments import get_comment, get_submitted_job_comment
+
+
+class MockIssueComment:
+    def __init__(self, body):
+        self.body = body
+
+
+@pytest.fixture
+def raise_exception():
+    with patch('github.PullRequest.PullRequest') as mock_pr:
+        instance = mock_pr.return_value
+        instance.get_issue_comments.side_effect = Exception()
+        instance.get_issue_comments.return_value = ()
+        yield instance
+
+
+@pytest.fixture
+def pr_no_comments():
+    with patch('github.PullRequest.PullRequest') as mock_pr:
+        instance = mock_pr.return_value
+        instance.get_issue_comments.return_value = ()
+        yield instance
+
+
+@pytest.fixture
+def pr_single_comment():
+    with patch('github.PullRequest.PullRequest') as mock_pr:
+        instance = mock_pr.return_value
+        instance.get_issue_comments.return_value = (MockIssueComment("foo"),)
+        yield instance
+
+
+# cases:
+#  - no comments exist
+#  - search string should be found
+#  - search string should not be found
+#  - calling get_issue_comments raises an Exception
+def test_get_comment_no_comment(pr_no_comments):
+    expected = None
+    actual = get_comment(pr_no_comments, "foo")
+    assert expected == actual
+
+
+def test_get_comment_found(pr_single_comment):
+    expected = MockIssueComment("foo").body
+    actual = get_comment(pr_single_comment, "foo").body
```

Comment:
no need to have a local `expected` variable:

```python
assert get_comment(pr_single_comment, "foo").body == "foo"
```

Reply:
I guess we need separate tests because of the fixtures being used...

Or maybe we can still collapse it with:

```python
def test_get_comment(pr_no_comments, pr_single_comment, raise_exception):
   ...

Reply:
> no need to have a local `expected` variable:
> 
> ```python
> assert get_comment(pr_single_comment, "foo").body == "foo"
> ```

Personally, I find the style

```python
expected = ...
actual = ...
assert expected == actual
```

much easier to read. For me it's a well-spent line of code. One could say the same about `actual`.

Reply:
> Should we have separate tests, or just one `test_get_comment` which tests different "use case"?
> 
> I somewhat prefer the latter, but it is sort of cosmetic. I would be helpful to have a consistent testing "style" (see `test_run_cmd`, which follows the latter approach)

Ack on the style. We shall use only one. Putting all fixtures into one test function may work (will test that). I was also looking into parametrising tests. Not sure if this would be easier with separate functions or if it doesn't matter.

There is one difference (don't know if it makes a _big_ difference in practice) which I believe is not cosmetic. If you have one assert per test function, all functions are run and they either fail or pass. If a function has multiple asserts they are all combined into one result.

Reply:
Yeah, that last bit is actually a good point. Maybe it's better to have many tiny tests, so you get a better view on what was broken. But then we need to also adopt the same style in our other test suites.

Reply:
No strong objection on keeping the more explicit/longer `expected/actual` style, I can see how that's easier to parse for humans (and more lines of code in the tests is not a bad thing in general if it helps with making the test easier to understand)